### ISSUE TITLE
Get ready for Leaflet 1.0

### ISSUE THREAD

**Issue by nmccready** [angular-ui/ui-leaflet#89]
_From @simison on July 15, 2015 11:24_

http://leafletjs.com/2015/07/15/leaflet-1.0-beta1-released.html

>The release sports a number of minor breaking API changes, and some of the plugins will have to be updated. To make the upgrade less painful, we’re doing a short beta cycle before the final 1.0 to allow users and plugin developers to catch up to the changes while we find and fix remaining issues.

:-)

_Copied from original issue: tombatossals/angular-leaflet-directive#851_

**Comment by nmccready** [comment]
_From @tombatossals on July 15, 2015 13:54_

Wow, great news. That would be a good moment for giving this directive a power up.

Thanks for reporting!

**Comment by nmccready** [comment]
_From @paambaati on August 4, 2015 4:56_

@tombatossals Is there a branch for this that we could check out/test/contribute?

**Comment by nmccready** [comment]
Referencing Mobile Draw Support . IE: This is another reason to get started on this.
https://github.com/Wildhoney/Leaflet.FreeDraw/issues/3

**Comment by nmccready** [comment]
I will be adding a leaflet 1.0.X branch to start working towards this. However I am not sure what the name of the branch should be.

leaflet-1.0.X to targets leaflet? Or just 1.0.X as we are working towards a 1.0.X ourselves. 

My only beef with making it 1.0.X is that going to leaflet 1.0.X might not be stable right away. But keeping the branch with 1.0.X and mapping to leaflet 1.0.X helps eliminate confusion.

Thoughts anyone?

**Comment by nmccready** [comment]
_From @vrubert on September 23, 2015 16:54_

Looks good @nmccready, I think too we need that branch because it will need some time to have the directive working with leaflet 1.0.x and all the modules.

BTW, the Leaflet.FreeDraw project is amazing!!

**Comment by nmccready** [comment]
@vrubert @tombatossals any thoughts on what you want the branch name to be?

**Comment by nmccready** [comment]
_From @simison on September 24, 2015 10:31_

Keeping the future version number of this package at the branch name makes more sense to me rather than following Leaflet versions. Trying to follow Leaflet versions and keeping them in pair with package, might prove hard in long term, doable only maybe for major versions.

I'd call it "1.0.0-with-leaflet-1.x", "1.0.0-WIP" or just "1.0.0"

**Comment by nmccready** [comment]
_From @tombatossals on September 24, 2015 17:56_

Hey, I would suggest  "1.0.0-wip" as @simison said, what do you think?

I haven't had time to take a look at leaflet-1.0-beta, I expect there will be no major API changes and 90% will work seamlessly. I hope so.

**Comment by nmccready** [comment]
Awesome I hope so too, what is WIP? Work In Progress? I am not sure if that is needed as all branches are WIP :grinning:  .

**Comment by nmccready** [comment]
Or is it WLP?

**Comment by nmccready** [comment]
hmm maybe just leaflet-1.X

**Comment by nmccready** [comment]
That way if we have any major changes (unseen foreseeable) we don't need to rename the branch.

**Comment by nmccready** [comment]
_From @tombatossals on September 24, 2015 18:9_

You're right, all branches are WIP :D 

Totally agree, better leaflet-1.X so no need to rename it later.

**Comment by nmccready** [comment]
cool I'll push the branch up

**Comment by nmccready** [comment]
_From @simison on September 24, 2015 21:42_

"WIP" suggests that it's currently in progress and shouldn't be expected to work. ;-)

leaflet-1.x is cool tho. Thanks for working on it!

**Comment by nmccready** [comment]
Working on this, and I can not find the leaflet-src.js dist file for master or 1.0.0-beta1

**Comment by nmccready** [comment]
See https://github.com/Leaflet/Leaflet/issues/3635

**Comment by nmccready** [comment]
Good lord, they made it incompatible with bower.. ie it is not standardized anymore. 

https://github.com/Leaflet/Leaflet/issues/1354

**Comment by nmccready** [comment]
Ok, so far we have a few alternatives

- bower will point to `"leaflet": "http://cdn.leafletjs.com/downloads/leaflet-1.0.0-b1.zip"` . The big problem with this is that it makes our library and leaflet dependencies an invalid mess where version number really means nothing.

- become a npm project and abandon bower

- fork leaflet to get the tags to have `dist/*.js` files again

This is a real problem with a lot of future headache. It would be nice if they could start using bower correctly again. But I am not confident this will happen.

**Comment by nmccready** [comment]
_From @simison on September 30, 2015 19:22_

Hmm, nope, normally they do have dist files for releases. (https://github.com/Leaflet/Leaflet/issues/3635#issuecomment-125168272)

It's just this beta which is lacking those files (https://github.com/Leaflet/Leaflet/issues/3635#issuecomment-123267747); it'll be fixed with the next version.

For WIP using the beta-1 zip path is fine.

**Comment by nmccready** [comment]
Yeah they need to do another release. However this still doesn't make me that confident in bower as a solution. This will probably be a headache in the future for sure. I vote for becoming npm centric.

**Comment by nmccready** [comment]
This brings up a good discussion though. It would be nice to not always be committing `dist/*.js` as it would keep the PR size very small and easier to review. I wonder if we should start doing something similar and only publish dist/*.js on tags/releases. Alternatively we could have a dist repo. Ugh.. all more work I don't feel like doing.

**Comment by nmccready** [comment]
With the `dist/*.js` ignored solution; this would only mess with people that target master. This would be tons of plunkers and many users.

**Comment by nmccready** [comment]
All the branch exists on the main repo. Feel free to start making the specs work. On a positive note the 604 example appears to work at least partially.

**Comment by nmccready** [comment]
_From @simison on September 30, 2015 19:58_

Bower will eventually die out as NPM evolves, I suppose. Anyway, for this sort of package it's very important to stay distribution channel agnostic and just keep supporting everything that's feasible. Bower is still hugely popular for good reasons.

Having a CI script to produce dist files sounds totally something this package should have. :-) Have a look at Leaflet's build scripts?

**Comment by nmccready** [comment]
There scripts are all using jake, I would prefer to look at grunt or gulp as I do not feel like learning another tool just for another tool sake.

**Comment by mobidev111** [comment]
@nmccready @elesdoar any indication when there will be a version ready for leaflet 1.0? what's the status - which things still need to be done?

**Comment by nmccready** [comment]
@elesdoar knows more than I do at the moment. I have not had time to work on this lately.

**Comment by weinspach** [comment]
@nmccready @elesdoar Any news on a leaflet 1.0 version? I have to use leaflet 1.0 for Esri vector basemaps which are not supported by leaflet < 1.0. Thanks.

**Comment by nmccready** [comment]
I am barely supporting this library at this point. I am mainly doing maintenance when needed. My main reasoning is I have come to the decision that angular is not appropriate for mapping. I could go into the details but I don't think it is worth it.

**Comment by mobidev111** [comment]
@nmccready Thanks for the feedback. I respect your decision.

There are quite a number of users who can't make this decision - e.g. we are bound to both angular and mapping.

For the the longterm we seek for maintainers to continue the project and ask the community for help. I'll create an issue for this.

For now: Could you start a basic migration to 1.0 to update the directive? Not the full fledged planned refactor, but just make it work with Leaflet 1.0. I understand that you are the the expert and in the know. We and probably many other users can assist with testing and so on..

**Comment by nmccready** [comment]
@elesdoar may have better answers as he has worked on major changes more recently

**Comment by elesdoar** [comment]
I was working on this compatibility and this is close to finishing, there are only some tests with some plugins, I will soon return to work on this project and I will advance these releases.

Please have patience, because we only work 3 people in the short time that we are remains free.

Thanks.

**Comment by weinspach** [comment]
Glad to hear that. Thank you.

**Comment by weinspach** [comment]
Ping :)
